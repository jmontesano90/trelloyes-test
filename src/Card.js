import React from 'react';
import './Card.css';

export default function Card(props) {
  const handleDelete = () =>{
      props.onDelete(props.cardId, props.listId);
  }

  return (
    <div className='Card'>
      <button
        onClick={handleDelete}
        type='button'
      >
        delete
      </button>
      <h3>{props.title}</h3>
      <p>{props.content}</p>
    </div>
  )
}