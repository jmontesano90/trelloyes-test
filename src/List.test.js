import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import Card from './Card'

let allCards = [
    {
        id: 1,
        title: "First Card",
        content: "Lorem ipsum"
    },

    {
        id: 2,
        title: "Second Card",
        content: "Lorem ipsum"
    },

    {
        id: 3,
        title: "Third Card",
        content: "Lorem ipsum"
    },
];

describe('List component', () =>{
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', ()  =>{
        const tree = renderer
            .create(
                <section className='List'>
                <header className='List-header'>
                  <h2>Second List</h2>
                </header>
                <div className='List-cards'>
                    <Card
                      key={allCards[2].id}
                      title={allCards[2].title}
                      content={allCards[2].content}
                    />
                  <button
                    type='button'
                    className='List-add-button'
                  >
                    + Add Random Card
                  </button>
                </div>
              </section>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});