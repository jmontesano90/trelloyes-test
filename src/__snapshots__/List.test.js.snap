// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second List
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Third Card
      </h3>
      <p>
        Lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
